gemet/thesaurus/import_spreadsheet.py

```diff
@@ -0,0 +1,304 @@
+                    status=PENDING,
+                    date_entered=timezone.now(),
+                )
+                print(u'Concept added: {}'.format(label))
+
+            self.concepts[label.lower()] = concept
+
+            concept.update_or_create_properties(property_values)
+
+            if is_new_concept:
+                # Create internal "searchText" property with the concatenated
+                # values from all other concept properties.
+                search_text = get_search_text(
+                    concept.id, 'en', PENDING, self.version
+                )
+                if search_text:
+                    search_text.save()
+
+    def _create_relations(self, sheet):
+
+        property_types = PropertyType.objects.filter(
+            name__in=['broader', 'group']
+        )
+
+        for i, row in enumerate(row_dicts(sheet)):
+
+            source_label = row.get("Term")  # aka prefLabel
+
+            for property_type in property_types:
+
+                # Look for columns specifying relationships
+                if property_type.name == 'broader':
+                    target_label = row.get("Broader concept")
+                elif property_type.name == 'group':
+                    target_label = row.get("Group")
+
+                if not target_label:
+                    # If it doesn't exist, there is no relation to be created
+                    print(
+                        (
+                            'Row {} has neither "broader" nor "group" columns.'
+                        ).format(i)
+                    )
+                    continue
+
+                source = self.concepts[source_label.lower()]
+                target = self._get_concept(target_label)
+                namespace = namespace_for(property_type.name)
+
+                if not target:
+                    print(
+                        'Creating inexistent concept: {}'.format(target_label)
+                    )
+                    code = get_new_code(self.concept_ns)
+                    target = Concept.objects.create(
+                        code=code,
+                        namespace=namespace,
+                        version_added=self.version,
+                        status=PENDING,
+                        date_entered=timezone.now(),
+                    )
+                    target.properties.create(
+                        status=PENDING,
+                        version_added=self.version,
+                        language_id='en',
+                        name='prefLabel',
+                        value=target_label,
+                    )
+
+                # target is broader of source
+                relation = Relation.objects.filter(
+                    source=source,
+                    target=target,
+                    property_type=property_type,
+                ).first()
+
+                if not relation:
+                    relation = Relation.objects.create(
```

Comment:
I don't understand. If the broader concept column has a value, than at this point I will already have created the relation. This would create a duplicate "broader" relationship. Am I missing something?